src/frequenz/channels/_merge.py

```diff
@@ -187,6 +191,21 @@ def consume(self) -> ReceiverMessageT_co:
 
         return self._results.popleft()
 
+    @override
+    def close(self) -> None:
+        """Close the receiver.
+
+        After calling this method, new messages will not be received.  Once the
+        receiver's buffer is drained, trying to receive a message will raise a
+        [`ReceiverStoppedError`][frequenz.channels.ReceiverStoppedError].
+        """
+        for task in self._pending:
+            if not task.done() and task.get_loop().is_running():
```

Comment:
Mmm, interesting the `task.get_loop().is_running()`. Was this to address warnings in tests or any other reason? I wonder if it is a good practice to do in general or not (and for a normal cleanup all tasks should be cancelled before stopping the loop).

Reply:
> Was this to address warnings in tests or any other reason?

Yes, I think I started adding it just for tests and I'm sure we'll find it in other places in this repo as well (unless you removed it).

> I wonder if it is a good practice to do in general or not (and for a normal cleanup all tasks should be cancelled before stopping the loop).

I agree, these could go away once we reach a point where removing them won't cause warning spam.